mergesort.js

```diff
@@ -1,22 +1,22 @@
 function mergesort(a){
-  
-  if (a.length == 1){
+
+  if (a.length === 1){
     return a;
   }
-  else if (a.length == 2){
+
+  if (a.length === 2){
     if (a[0] > a[1]){
       return [a[1], a[0]];
     }
-    else{
-      return a;
-    }
+
+    return a;
   }
 
-  var p = parseInt(a.length/2);
-  var m1 = mergesort(a.slice(0, p));
-  var m2 = mergesort(a.slice(p));
+  const p = parseInt(a.length / 2);
+  let m1 = mergesort(a.slice(0, p));
+  let m2 = mergesort(a.slice(p));
 
-  var ret = [];
+  let ret = [];
 
   while(1){
     if(m1.length > 0 && m2.length > 0){
```

Comment:
White space between "if" and "(" required. Same with while loop.

Reply:
i didn't want to reformat this because i wasn't sure if coding standards matter at all.
i can however apply the [google javascript style guide](https://google.github.io/styleguide/jsguide.html) if you want me to.

Reply:
It is common code style convention.
The white space between if/case/all loop is used to clearly understand, that it's not a function.

For example:

if (age > 10) // here's an if-statement

ifAge(10) // here's a function

It's like a basic rule for all programming languages.

Reply:
Thats not true. 
Spacing varies depending on coding standard & language.
`if` is a keyword in most languages and will never be used as a function / method name.

If this spacing wasn't allowed, syntax errors would be thrown.

I applied the style guide mentioned above.

Reply:
Of course, you cannot create a function named if, but this is a visual difference for quick understanding.
By the way, I'll do some research to find out. Maybe white space was important in the past